networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BeaconBlocksByRootMessageHandler.java

```diff
@@ -38,6 +45,10 @@ public void onIncomingMessage(
       final ResponseCallback<SignedBeaconBlock> callback) {
     LOG.trace(
         "Peer {} requested BeaconBlocks with roots: {}", peer.getId(), message.getBlockRoots());
+    if (message.getBlockRoots().size() > Constants.MAX_REQUEST_BLOCKS) {
```

Comment:
Is it possible situation? Shouldn't we fail on SSZ deserialization?
Nothing criminal of course, but looks strange to check here the condition which should be enforced implicitly by the `SSZList` max size.